### PR TITLE
Adds State Management capabilities to Macaw

### PR DESCRIPTION
**Pull request by gzhihongwei**
- Implements [this](https://trello.com/c/D50P1uMa/1-state-manager-to-save-and-load-any-state-variables) card
  - Added schema as `schema.ts`
    - Extends the `Message` class with `nlp_pipeline`, `user_attributes`, `actions`, and `dialog_state_tracking` dictionaries/classes
   - Added `CurrentAttributes`, `UserAttributes`, and `Singleton` classes

Need to add documentation of how to access specific attributes, but most of the state management is more dependent on the actions utilizing the `Message` class.

### CODE REVIEW

File: schema.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+interface Message {
```
Comment (Amit-GH) [inline schema.ts:1]: How is this file used?

**Review by Amit-GH — COMMENTED** [review]
Very nice changes. This will help a lot. 

I couldn't run this change locally due to python and other issues. I have created a separate PR to address those fix. Look into that and then update this PR. Make sure you can run the application and get responses to some basic questions.

File: schema.ts (hunk 1 above)
Comment (gzhihongwei) [reply schema.ts:1]: It's not used per se, it was the best way for me to visualize what the schema is. I can put it in the README later after it is finalized.

Comment (Amit-GH) [reply schema.ts:1]: Okay. That'll be better. In the current forms it seems as if the Message interface is handled through this file which isn't true.

**Review by Amit-GH — APPROVED** [review]

**Pull request merged**